### PR TITLE
sig int 656 - fix pr

### PR DESCRIPTION
**Pull request by lsynopsys**

### CODE REVIEW

File: src/synopsys-action/tools-parameter.ts [hunk 1]
```diff
@@ -107,7 +108,7 @@ export class SynopsysToolsParameter {
       covData.data.project.branch = {name: inputs.COVERITY_BRANCH_NAME}
     }
 
-    if (inputs.COVERITY_AUTOMATION_PRCOMMENT) {
+    if (parseBoolean(inputs.COVERITY_AUTOMATION_PRCOMMENT)) {
       covData.data.github = this.setGithubData()
```
Comment (sig-tithi) [inline src/synopsys-action/tools-parameter.ts:112]: can we put a log for coverity also

File: src/synopsys-action/tools-parameter.ts [hunk 2]
```diff
@@ -185,15 +186,16 @@ export class SynopsysToolsParameter {
...
     } else {
       // Disable fix pull request for adapters
       blackduckData.data.blackduck.automation.fixpr = false
     }
 
-    if (inputs.BLACKDUCK_AUTOMATION_PRCOMMENT) {
+    if (parseBoolean(inputs.BLACKDUCK_AUTOMATION_PRCOMMENT)) {
       info('Blackduck Automation comment is enabled')
       blackduckData.data.github = this.setGithubData()
       blackduckData.data.blackduck.automation.prcomment = Boolean(inputs.BLACKDUCK_AUTOMATION_PRCOMMENT)
```
Comment (sig-tithi) [inline src/synopsys-action/tools-parameter.ts:201]: if BLACKDUCK_AUTOMATION_PRCOMMENT is not set do we need to pass it as false to bridge or we won't pass it at all like we have for blackduckData.data.blackduck.automation.fixpr

Comment (lsynopsys) [reply src/synopsys-action/tools-parameter.ts]: done

File: src/synopsys-action/utility.ts [hunk 3]
```diff
@@ -27,3 +27,10 @@ export async function cleanupTempDir(tempDir: string): Promise<void> {
 export function checkIfGithubHostedAndLinux(): boolean {
   return String(process.env['RUNNER_NAME']).includes('Hosted Agent') && (process.platform === 'linux' || process.platform === 'darwin')
 }
+
+export function parseToBoolean(value: string | boolean): boolean {
+  if (value === true || (value !== null && value !== '' && value.toString().toLowerCase() === 'true')) {
```
Comment (sig-tithi) [inline src/synopsys-action/utility.ts:32]: Why are we checking value === true  before null check?

**Review by sig-tithi — APPROVED** [review]

File: src/synopsys-action/utility.ts [hunk 4]
```diff
@@ -27,3 +27,10 @@ export async function cleanupTempDir(tempDir: string): Promise<void> {
 export function checkIfGithubHostedAndLinux(): boolean {
   return String(process.env['RUNNER_NAME']).includes('Hosted Agent') && (process.platform === 'linux' || process.platform === 'darwin')
 }
+
+export function parseToBoolean(value: string | boolean): boolean {
```
Comment (kishorikumar) [inline src/synopsys-action/utility.ts:31]: What will happen, if we pass "truee" or any other value apart from "true"/"false"? Shall we consider that false only or we show show error/warning?

**Review by kishorikumar — CHANGES_REQUESTED** [review]

File: src/synopsys-action/utility.ts (hunk 4 above)
Comment (lsynopsys) [reply src/synopsys-action/utility.ts:31]: we should consider that as false itself

**Pull request merged**